### PR TITLE
task solution

### PR DESCRIPTION
**Pull request by BeginerAlex**
[DEMO LINK](https://begineralex.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +17,113 @@ export const goodsFromServer = [
...
+    return prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortParams, setSortParams] = useState<SortParams>({
+    sortField: SortType.SORT_BY_ALPHABET,
+    reverse: false,
```
Comment (Viktor-Kost) [inline src/App.tsx:56]: https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig
```suggestion
    isReversed: false,
```

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +17,113 @@ export const goodsFromServer = [
...
+            { 'is-light': sortParams.sortField !== SortType.SORT_BY_ALPHABET })
+          }
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => {
+            setSortParams({ sortField: SortType.SORT_BY_LENGTH, reverse: false});
+          }}
```
Comment (Viktor-Kost) [inline src/App.tsx:86]: move to an own function. this function should accept `key` and `value` arguments, spread current params object and set a new one
Example:
```
  const handleSortClick = (key, value) => {
    setSortParams(prev => ({...prev, [key]: value}));
  }
```

File: src/App.tsx [hunk 3]
```diff
@@ -15,48 +17,113 @@ export const goodsFromServer = [
...
+          className={cn('button is-success',
+            { 'is-light': sortParams.sortField !== SortType.SORT_BY_LENGTH })
+          }
         >
           Sort by length
         </button>
 
         <button
+          onClick={() => {
+            setSortParams({ ...sortParams, reverse: !sortParams.reverse });
```
Comment (Viktor-Kost) [inline src/App.tsx:97]: the same

File: src/App.tsx [hunk 4]
```diff
@@ -15,48 +17,113 @@ export const goodsFromServer = [
...
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortParams, setSortParams] = useState<SortParams>({
+    sortField: SortType.SORT_BY_ALPHABET,
+    reverse: false,
+  });
```
Comment (Viktor-Kost) [inline src/App.tsx:57]: move a state object to a variable. You can use this variable here and at reset function as well

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Nice!
Some minor fixes needed

**Review by etojeDenys — CHANGES_REQUESTED** [review]
try to fix the demo link and then re-request me

**Pull request closed without merging**